### PR TITLE
[Tests] `jsx-indent`, `jsx-one-expression-per-line`: add passing test cases

### LINKED ISSUE #2318: [react/jsx-one-expression-per-line] moves text to the beginning of line
Hi! I tried to use "react/jsx-one-expression-per-line" rule, but its behaviour is very strange and I don't know how can I fix it.

Sandbox : https://codesandbox.io/s/gatsby-starter-default-f7u3w , please do `npm run lint` in terminal

My config `.eslintrc.json`
```json
{
   "parserOptions": {
        "ecmaVersion": 10,
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "plugins": [
        "react"
    ],
    "extends": [
        "eslint:recommended",
        "plugin:react/recommended"
    ],
    "rules": {
        "react/jsx-one-expression-per-line": [2, { "allow": "none" }]
    }
}
```

Before running `eslint --fix`
```html
    <h1>Hi people</h1>
    <p>Welcome to your new Gatsby site.</p>
    <p>Now go build something great.</p>
```
After applying eslint
```html
    <h1>
Hi people
</h1>
    <p>
Welcome to your new Gatsby site.
</p>
    <p>
Now go build something great.
</p>
```

### PR DESCRIPTION
**Pull request by ROSSROSALES**
Increased test coverage for jsx-one-expression-per-line.js 

Fixes #2318

### CODE REVIEW

**Comment by ROSSROSALES** [comment]
Passing test cases for issue #2318

**Comment by ljharb** [comment]
I think we should add the exact test code from the issue:
```js
  <Layout>
    <SEO title="Home" />
    <h1>{"Hi people"}<button/></h1>
    <p>Welcome to your new Gatsby site.</p>
    <p>Now go build something great.</p>
    <h1>Hi people<button/></h1>
    <div style={{ maxWidth: `300px`, marginBottom: `1.45rem` }}>
      <Image />
    </div><Link to="/page-2/">Go to page 2</Link>
  </Layout>
```

**Comment by ljharb** [comment]
Yep, exactly! I'd expect the same test input to be added to both rules' test files, each covering the autofix output of the relevant rule.

File: tests/lib/rules/jsx-one-expression-per-line.js [hunk 1]
```diff
@@ -1389,5 +1389,185 @@ a
       ],
       parserOptions,
     },
+    {
+    // Would be nice to handle in one pass, but multipass works fine.
```
Comment (ljharb) [inline tests/lib/rules/jsx-one-expression-per-line.js:1393]:
```suggestion
    // TODO: handle in a single pass
```

File: tests/lib/rules/jsx-one-expression-per-line.js [hunk 2]
```diff
@@ -1389,5 +1389,185 @@ a
...
+      `,
+      errors: [
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'button' },
+        },
+      ],
+      parserOptions,
+    },
+    // Would be nice to handle in one pass, but multipass works fine.
```
Comment (ljharb) [inline tests/lib/rules/jsx-one-expression-per-line.js:1440]:
```suggestion
    // TODO: handle in a single pass (see above)
```

File: tests/lib/rules/jsx-one-expression-per-line.js [hunk 3]
```diff
@@ -1389,5 +1389,185 @@ a
...
```
Comment (ljharb) [inline tests/lib/rules/jsx-one-expression-per-line.js:1516]:
```suggestion
    // TODO: handle in a single pass
```

File: tests/lib/rules/jsx-one-expression-per-line.js [hunk 4]
```diff
@@ -1389,5 +1389,185 @@ a
...
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'Go to page 2' },
+        },
+      ],
+      parserOptions,
+    },
   ]),
-});
+});
```
Comment (ljharb) [inline tests/lib/rules/jsx-one-expression-per-line.js:1573]: trailing newline

**Review by ljharb — COMMENTED** [review]
Looks great!

**Review by ljharb — APPROVED** [review]

**Pull request merged**